src/lib/core/errinj.h

```diff
@@ -176,6 +176,7 @@ void errinj_set_with_environment_vars(void);
 
 #ifdef NDEBUG
 #  define ERROR_INJECT(ID, CODE)
+#  define ERROR_INJECT_COND(ID, TYPE, COND, CODE)
```

Comment:
Why not just ERROR_INJECT_NONZERO? Are you see cases, where we actually need an arbitrary condition?

BTW, why `ERRINJ_STDIN_ISATTY` is not boolean?

Reply:
> Why not just ERROR_INJECT_NONZERO? Are you see cases, where we actually need an arbitrary condition?

Yes, see here:

- https://github.com/tarantool/tarantool/blob/master/src/box/wal.c#L248-L252 
- https://github.com/tarantool/tarantool/blob/master/src/lib/core/fiber.c#L202-L205

> BTW, why `ERRINJ_STDIN_ISATTY` is not of the boolean type?

Because it may have 3 different values: -1 (default, enabled), 0 and 1.

Reply:
Seems legit.